pkg/kubelet/prober/manager.go

```diff
@@ -207,6 +207,23 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *api.PodStatus) {
 		}
 		podStatus.ContainerStatuses[i].Ready = ready
 	}
+	// init containers are ready if they have exited with success or if a readiness probe has
+	// succeeded.
+	for i, c := range podStatus.InitContainerStatuses {
```

Comment:
If we are not going to use/test this code path, my gut feeling is that it'll probably get outdated sooner or later. The fact that "ready" means differently for init containers is also slightly misleading. Why can't we just check the terminated exitcode when needed?